src/core_modules/capture-core/components/Pages/NewRelationship/RegisterTei/DataEntry/RegisterTeiDataEntry.container.js

```diff
@@ -1,23 +1,20 @@
 // @flow
+import { compose } from 'redux';
 import { connect } from 'react-redux';
 import DataEntry from './RegisterTeiDataEntry.component';
 import withErrorMessageHandler from '../../../../../HOC/withErrorMessageHandler';
-import withLoadingIndicator from '../../../../../HOC/withLoadingIndicator';
 
 const mapStateToProps = (state: ReduxState) => ({
     showDataEntry: state.newRelationshipRegisterTei.orgUnit,
-    ready: !state.newRelationshipRegisterTei.dataEntryIsLoading,
     error: state.newRelationshipRegisterTei.dataEntryError,
     programId: state.newRelationshipRegisterTei.programId,
 });
 
 const mapDispatchToProps = () => ({});
 
-// $FlowFixMe[missing-annot] automated comment
-export const RegisterTeiDataEntry = connect(mapStateToProps, mapDispatchToProps)(
-    withLoadingIndicator()(
```

Comment:
With this change I am handling the loading on the DataEntry level. This is arguably almost the same so the change doesnt serve as an improvement. 

To understand why this change is here we will need to think like that. 
Now that the component itself triggers its initialisation, meaning adding the relevant information to redux we cant have the loading indicator depend on the redux state since it hasnt been yet initialised. Thus I move it on a lower level :)